content/tutorials/reactor-firehose-kinesis.textile

```diff
@@ -0,0 +1,602 @@
+** *Partition Key*: This facility enables you to stream messages to different shards, based on a "routing key":https://faqs.ably.com/what-is-the-format-of-the-routingkey-for-an-amqp-or-kinesis-reactor-rule. As your stream contains only a single shard, enter @1@ here.
+    <a href="/images/tutorials/reactor-firehose-kinesis/firehose-kinesis-edit-rule-1.png" target="_blank">
+      <img src="/images/tutorials/reactor-firehose-kinesis/firehose-kinesis-edit-rule-1.png" style="width: 100%"
+      title="Click to enlarge" alt="Configuring your Reactor rule">
+    </a>
+    <a href="/images/tutorials/reactor-firehose-kinesis/firehose-kinesis-edit-rule-2.png" target="_blank">
+      <img src="/images/tutorials/reactor-firehose-kinesis/firehose-kinesis-edit-rule-2.png" style="width: 100%"
+      title="Click to enlarge" alt="Configuring your Reactor rule">
+    </a>
+
+Test your rule before proceeding to the next step, to ensure that you have the necessary permissions configured correctly. Click the Test rule button then, in the dialog that displays, click Run test now:
+    <a href="/images/tutorials/reactor-firehose-kinesis/firehose-kinesis-test-rule.png" target="_blank">
+      <img src="/images/tutorials/reactor-firehose-kinesis/firehose-kinesis-test-rule.png" style="width: 100%"
+      title="Click to enlarge" alt="Testing your Reactor rule">
+    </a>
+
+h2. Step 4: Create your server
+
+In this step you will create a Node.js server to test your Reactor Firehose integration with Amazon Kinesis. 
+
+First, create a directory for your application and change into it:
+
+```[sh]
+mkdir firehose-kinesis-demo; cd firehose-kinesis-demo
+```
+
+Create two subdirectories within your application directory, to store the static HTML, JavaScript, and CSS files:
+
+```[sh]
+mkdir public views
+```
+
+Then, execute @npm init@ to create a Node.js project. For @entry point@ enter @server.js@ and accept the defaults for the other values.
+
+Install the dependencies you will require for this project:
+** @ably@: The Ably client library.
+** @express@: A web application framework for Node.js.
+** @dotenv@: A module that makes it easy to store and read configuration settings from a @.env@ file.
+** @random-number@: A helper module you will use to create random sensor data.
+    
+```[sh]
+npm install ably express dotenv random-number
+```
+
+Create a @.env@ file in your application directory to store configuration settings. Replace @your_ably_api_key@ with your app's API key that you generated in Step 3. You can also view your app's API key  in the Ably dashboard:
+
+```[plain]
+ABLY_API_KEY=your_ably_api_key
+NUM_SENSORS=5 
+PORT=5000  
+```
+
+@NUM_SENSORS@ is the number of IoT devices your application will simulate, each of which will generate a random temperature and humidity reading. Set this value to five initially.
```

Comment:
A minor thing, but I'd write this as `5` since it's in the code sample above.

Reply:
Done